src/app/services/wallet.service.ts

```diff
@@ -41,60 +42,76 @@ export class WalletService {
     });
   }
 
-  decrypt(passphrase: string, walletEnckey: string): Observable<boolean> {
-    let result = new BehaviorSubject<boolean>(null);
-    let selectedWalletId: string;
-    this.getSelectedWallet().subscribe(
-      (selectedWallet) => (selectedWalletId = selectedWallet.id)
-    );
+  async decrypt(passphrase: string, walletEnckey: string) {
+    var wallet_found: Wallet = await new Promise((resolve) => {
+      this.getSelectedWallet().subscribe((wallet) => {
+        resolve(wallet);
+      });
+    });
```

Comment:
Maybe use `toPromise`?
```suggestion
const wallet_found = await this.getSelectedWallet().toPromise();
```
p.s. code may not compile

Reply:
ok, i will try